pkg/client/testing/core/fixture.go

```diff
@@ -49,153 +58,158 @@ type ObjectScheme interface {
 	runtime.ObjectTyper
 }
 
-// ObjectReaction returns a ReactionFunc that takes a generic action string of the form
-// <verb>-<resource> or <verb>-<subresource>-<resource> and attempts to return a runtime
-// Object or error that matches the requested action. For instance, list-replicationControllers
-// should attempt to return a list of replication controllers. This method delegates to the
-// ObjectRetriever interface to satisfy retrieval of lists or retrieval of single items.
-// TODO: add support for sub resources
-func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
+// ObjectReaction returns a ReactionFunc that applies core.Action to
+// the given registry.
+func ObjectReaction(registry Registry, mapper meta.RESTMapper) ReactionFunc {
 	return func(action Action) (bool, runtime.Object, error) {
+		ns := action.GetNamespace()
 		resource := action.GetResource()
+
 		kind, err := mapper.KindFor(resource)
+		if err != nil {
+			return false, nil, fmt.Errorf("unrecognized action for resource %s: %s", resource, err)
+		}
+
 		// This is a temporary fix. Because there is no internal resource, so
 		// the caller has no way to express that it expects to get an internal
 		// kind back. A more proper fix will be directly specify the Kind when
 		// build the action.
 		kind.Version = resource.Version
-		if err != nil {
-			return false, nil, fmt.Errorf("unrecognized action %s: %v", action.GetResource(), err)
-		}
 
-		// TODO: have mapper return a Kind for a subresource?
-		switch castAction := action.(type) {
-		case ListAction:
+		// Here and below we need to switch on implementation types,
+		// not on interfaces, as some interfaces are identical
+		// (e.g. UpdateAction and CreateAction), so if we use them,
+		// updates and creates end up matching the same case branch.
+		switch typedAction := action.(type) {
+
+		case ListActionImpl:
 			kind.Kind += "List"
-			resource, err := o.Kind(kind, "")
-			return true, resource, err
+			obj, err := registry.Get(kind, ns, "")
+			return true, obj, err
 
-		case GetAction:
-			resource, err := o.Kind(kind, castAction.GetName())
-			return true, resource, err
+		case GetActionImpl:
+			obj, err := registry.Get(kind, ns, typedAction.GetName())
+			return true, obj, err
 
-		case DeleteAction:
-			resource, err := o.Kind(kind, castAction.GetName())
-			return true, resource, err
+		case CreateActionImpl:
+			objMeta, err := meta.Accessor(typedAction.GetObject())
+			if err != nil {
+				return true, nil, err
+			}
+			if typedAction.GetSubresource() == "" {
+				err = registry.Add(typedAction.GetObject())
+			} else {
+				// Handle subresource creation as an update on the
```

Comment:
Can you model this as a separate method that takes the subresource and the object and must return the correct value to you?  Some subresource posts require special handling.  I can certainly see making a standard implementation for handling "status" since its so common.